### PR TITLE
Introduce new Query interface

### PR DESCRIPTION
**Pull request by aelred**
The `Query` interface is shared between all query types and contains any common methods, such as `execute` and `resultString`.

### CODE REVIEW

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/InsertQueryImpl.java [hunk 1]
```diff
@@ -81,9 +81,20 @@ public InsertQuery withGraph(MindmapsGraph graph) {
     }
 
     @Override
-    public void execute() {
+    public Void execute() {
```
Comment (fppt) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/InsertQueryImpl.java]: Why the change to `Void` ?

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/ComputeQueryImpl.java [hunk 2]
```diff
@@ -68,7 +65,11 @@ public Object execute() throws ExecutionException, InterruptedException {
                 return analytics.degrees();
             }
             case "degreesAndPersist": {
```
Comment (fppt) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/ComputeQueryImpl.java]: Hard coded strings ? Can we not add this to one our many enums ? Hard coded strings make me cry.

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/AskQueryImpl.java [hunk 3]
```diff
@@ -38,11 +41,25 @@
     }
 
     @Override
-    public boolean execute() {
+    public Boolean execute() {
```
Comment (fppt) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/AskQueryImpl.java]: Is this change also because of the generic on the interface ?

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/InsertQueryImpl.java (hunk 1 above)
Comment (aelred) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/InsertQueryImpl.java]: Yes, that's exactly the reason.

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/ComputeQueryImpl.java (hunk 2 above)
Comment (aelred) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/ComputeQueryImpl.java]: This was just to get analytics in quickly. We're eventually going to support custom "compute" methods, which will probably go in some kind of map.

File: mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/AskQueryImpl.java (hunk 3 above)
Comment (aelred) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/internal/query/AskQueryImpl.java]: 👍
